### PR TITLE
test: add e2e tests for viewer

### PR DESCRIPTION
**Pull request by kirillovzah**

### CODE REVIEW

File: src/pages/viewer-app/e2e.viewer-app.spec.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    await page.waitForSelector('#control-panel');
+    await page.select('#base-map', 'Light');
+    const controlOptions = [
+      'New York',
+      'San Francisco v1.6',
+      'San Francisco v1.7',
+    ];
+
+    for await (const item of controlOptions) {
+      expect(await page.select('#tileset', item)).toEqual([item]);
```
Comment (dryabinin-actionengine) [inline src/pages/viewer-app/e2e.viewer-app.spec.ts:25]: Will be good to detect that tileset really has been changed not only in selector.
WE can't detect it in canvas but, memory stats widget displays tcurrent ileset url.
You can check it.

File: src/pages/viewer-app/e2e.viewer-app.spec.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    await page.click('#change-terrain');
+  }, 10000);
+
+  it('Contains building explorer', async () => {
```
Comment (dryabinin-actionengine) [inline src/pages/viewer-app/e2e.viewer-app.spec.ts:31]: Please add tests for internal check boxes

File: src/pages/viewer-app/e2e.viewer-app.spec.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      'New York',
+      'San Francisco v1.6',
+      'San Francisco v1.7',
+    ];
+
+    for await (const item of controlOptions) {
+      expect(await page.select('#tileset', item)).toEqual([item]);
+    }
+
+    await page.click('#change-terrain');
```
Comment (dryabinin-actionengine) [inline src/pages/viewer-app/e2e.viewer-app.spec.ts:28]: Check if toogle is changing state from 'active' to 'not active' and vice versa.

**Review by dryabinin-actionengine — CHANGES_REQUESTED** [review]

**Comment by dryabinin-actionengine** [comment]
Good Job!
Please fix conflicts and fix my comments.

File: src/pages/viewer-app/e2e.viewer-app.spec.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    for await (const [index, item] of controlOptions.entries()) {
+      expect(await page.select("#tileset", item)).toEqual([item]);
+      expect(
+        await page.$eval("#stats-panel", e =>
+          e.lastChild.firstChild.textContent.slice(2)
+        )
+      ).toBe(controlUrls[index]);
+    }
+
+    await page.click("#change-terrain");
```
Comment (belom88) [inline src/pages/viewer-app/e2e.viewer-app.spec.ts:39]: I don't like creating new props for testing. You can use more long selector here `#terrain-layer-switch>span`

File: src/components/toogle-switch/toggle-switch.tsx [hunk 5]
```diff
@@ -71,9 +71,10 @@ const Slider = styled.span`
 export const ToggleSwitch = ({
   checked,
   onChange,
-  name = "",
-  id = "",
-  title = "",
+  name = '',
+  id = '',
+  sliderId = '',
```
Comment (belom88) [inline src/components/toogle-switch/toggle-switch.tsx:76]: I don't like this new prop. It is added only for testing that is not good. You can use more difficult selector instead:
`#[parentTogleSwitchId]>span`

File: src/pages/dashboard/e2e.dashboard.spec.ts [hunk 6]
```diff
@@ -84,4 +84,4 @@ describe("Dashboard", () => {
     const dashboardCanvas = await page.$$("#dashboard-app");
     expect(dashboardCanvas).toBeDefined();
   });
-});
+});
```
Comment (belom88) [inline src/pages/dashboard/e2e.dashboard.spec.ts:87]: Add new line in the end of the file

**Review by belom88 — CHANGES_REQUESTED** [review]

File: src/components/toogle-switch/toggle-switch.tsx [hunk 7]
```diff
@@ -84,10 +84,10 @@ export const ToggleSwitch = ({
           name={name}
           checked={checked}
           title={title}
-          onChange={onChange}
+          onChange={onChange}
```
Comment (dryabinin-actionengine) [inline src/components/toogle-switch/toggle-switch.tsx:87]: Remove extra space.

**Review by dryabinin-actionengine — CHANGES_REQUESTED** [review]

File: jest-puppeteer.config.js [hunk 8]
```diff
@@ -10,4 +10,4 @@ module.exports = {
     port: 3000,
     launchTimeout: 180000,
   },
-};
+};
```
Comment (belom88) [inline jest-puppeteer.config.js:13]: Why this file is in the changes?

**Review by belom88 — CHANGES_REQUESTED** [review]

**Comment by dryabinin-actionengine** [comment]
Will close this PR because this branch was crated from fork repo and now PR is open from original repo here - https://github.com/visgl/loaders.gl-showcases/pull/30

**Pull request closed without merging**